corehq/motech/repeaters/tasks.py

```diff
@@ -30,40 +35,27 @@
 )
 def check_repeaters():
     start = datetime.utcnow()
-    cutoff = start + CHECK_REPEATERS_INTERVAL
+    six_hours_later = start + timedelta(seconds=six_hours_sec)
 
-    # Timeout for slightly less than periodic check
+    # Long timeout to allow all waiting repeat records to be iterated
     check_repeater_lock = get_redis_lock(
```

Comment:
I think this lock is no longer necessary. If only one process is running, then it's definitely not necessary. If there are two or more, and multiple processes call `record.can_enqueue()` for the same record at the same moment, only one will return true, and all others will fail with `ResourceConflict` and return false.

Reply:
That sounds right, but wouldn't dropping the lock result in churn that using it avoids?

Reply:
Context from offline discussion: we decided that it might be better to measure first and then drop the lock later after verifying it's no longer needed. Do you want to add datadog monitoring of things like the _number of records_, _length of time spent in the loop_, and _the number of times the lock cannot be acquired_ in this PR or a follow-up?